lib/galaxy/datatypes/binary.py

```diff
@@ -462,6 +469,53 @@ def to_archive(self, trans, dataset, name=""):
         file_paths.append(dataset.metadata.bam_index.file_name)
         return zip(file_paths, rel_paths)
 
+
+    def get_chunk(self, trans, dataset, offset=0, ck_size=None):
+        bamfile = pysam.AlignmentFile(dataset.file_name, "rb")
+        ck_size = 100 # 100 lines
+        ck_data=""
+        lineNumber=0
+        if(offset == 0):
+            ck_data = bamfile.text
```

Comment:
We probably want to get the header only in the first chunk (i.e if offset is 0), since the header can be multiple thousand lines long.